ocrd_network/ocrd_network/processor_server.py

```diff
@@ -0,0 +1,205 @@
+            response_model=PYJobOutput,
+            response_model_exclude_unset=True,
+            response_model_exclude_none=True
+        )
+
+    async def startup(self):
+        await initiate_database(db_url=self.db_url)
+        DBProcessorJob.Settings.name = self.processor_name
+
+    async def get_processor_info(self):
+        if not self.ocrd_tool:
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail=f'Empty or missing ocrd_tool'
+            )
+        return self.ocrd_tool
+
+    # Note: The Processing server pushes to a queue, while
+    #  the Processor Server creates (pushes to) a background task
+    async def create_processor_job_task(self, job_input: PYJobInput, background_tasks: BackgroundTasks):
+        validate_job_input(self.log, self.processor_name, self.ocrd_tool, job_input)
+        job_input = await validate_and_resolve_mets_path(self.log, job_input, resolve=True)
+
+        job_id = generate_id()
+        job = DBProcessorJob(
+            **job_input.dict(exclude_unset=True, exclude_none=True),
+            job_id=job_id,
+            processor_name=self.processor_name,
+            state=StateEnum.queued
+        )
+        await job.insert()
+
+        execution_failed = False
+        start_time = datetime.now()
+        await db_update_processing_job(
+            job_id=job_id,
+            state=StateEnum.running,
+            start_time=start_time
+        )
+        try:
+            background_tasks.add_task(
+                run_single_execution,
+                ProcessorClass=self.ProcessorClass,
+                executable=self.processor_name,
+                abs_path_to_mets=job.path_to_mets,
+                input_file_grps=job.input_file_grps,
+                output_file_grps=job.output_file_grps,
+                page_id=job.page_id,
+                parameters=job.parameters
+            )
+        except Exception as error:
+            self.log.debug(f"processor_name: {self.processor_name}, path_to_mets: {job.path_to_mets}, "
+                           f"input_grps: {job.input_file_grps}, output_file_grps: {job.output_file_grps}, "
+                           f"page_id: {job.page_id}, parameters: {job.parameters}")
+            self.log.error(error)
+            execution_failed = True
+        end_time = datetime.now()
+        exec_duration = calculate_execution_time(start_time, end_time)
+        job_state = StateEnum.success if not execution_failed else StateEnum.failed
+        await db_update_processing_job(
+            job_id=job_id,
+            state=job_state,
+            end_time=end_time,
+            exec_time=f'{exec_duration} ms'
+        )
+        return job.to_job_output()
+
+    def get_ocrd_tool(self):
+        if self.ocrd_tool:
+            return self.ocrd_tool
+        if self.ProcessorClass:
+            ocrd_tool = self.ProcessorClass(workspace=None, version=True).ocrd_tool
```

Comment:
You're right. I just wanted to find a cleaner and more direct way to get the `ocrd_tool` and the `version` from the `ProcessorClass` directly. It was working on the surface so I did not dig into the details. Have already reverted that back to reading a stdout and parsing that instead as a temporal solution.

Reply:
I am resolving this. Should this continue in a separate discussion?

> Getting the .ocrd_tool from non-processing context is a good idea though. We could work with show_version=True (instead of version=True), which only has the side effect of printing the version to stdout (which can be intercepted or allowed).
Another option is to use dump_json=True, but not to access the .ocrd_tool member afterwards, but directly for its stdout.
Or we implement a new, tailored non-processing context in the constructor, say get_json=True, which does a return instead of a print – but that might not be allowed with Python.

Reply:
Not necessary – you already resolved all that with your changes.